flow-server/src/main/java/com/vaadin/flow/router/QueryParameters.java

```diff
@@ -189,6 +192,40 @@ public Map<String, List<String>> getParameters() {
         return parameters;
     }
 
+    /**
+     * Returns query parameter values mapped with the given key.
+     * <p>
+     * Example: Calling the method with key "one" for a parameters like
+     * {@code https://example.com/?one=1&two=2&one=3} will result in the
+     * corresponding list: {@code [1, 3]}
+     *
+     * @param key
+     *            the key of query parameters to fetch
+     * @return query parameters or null if there are no parameters with the
+     *         given key
+     */
+    public List<String> getParameter(String key) {
+        return parameters.get(key);
+    }
+
+    /**
+     * Returns the first query parameter values mapped with the given key.
+     * <p>
+     * Example: Calling with key value "one" with
+     * {@code https://example.com/?one=1&two=2&one=3} will return {@code 1}
+     *
+     * @param key
+     *            the key of query parameters to fetch
+     * @return query parameter value
+     */
+    public Optional<String> getSingleParameter(String key) {
+        try {
+            return Optional.ofNullable(parameters.get(key).get(0));
+        } catch (Exception e) {
+            return Optional.empty();
+        }
```

Comment:
Can this method be rewritten to avoid using exceptions for its logic?
Something like
```suggestion
        parameters.getOrDefault(key, Collections.emptyList()).stream().findFirst()
```
Or just get the parameter and check for null and empty